File: billing-service/src/main/java/com/petclinic/billing/datalayer/Bill.java
package com.petclinic.billing.datalayer;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Field;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Bill {

    @Id
    private String id;
    private String billId;
    private String customerId;
    private String ownerFirstName;
    private String ownerLastName;
    private String visitType;
    private String vetId;
    private String vetFirstName;
    private String vetLastName;
    private LocalDate date;
    private BigDecimal amount;
    private BigDecimal taxedAmount;
    private BigDecimal interest;
    private BillStatus billStatus;
    private LocalDate dueDate;
    @Builder.Default
    private boolean interestExempt = false;
    @Field("archive")
    private Boolean archive = false;
}

File: petclinic-frontend/src/features/bills/models/Bill.ts
export interface Bill {
  billId: string;
  customerId: string;
  ownerFirstName: string;
  ownerLastName: string;
  visitType: string;
  vetId: string;
  vetFirstName: string;
  vetLastName: string;
  date: string;
  amount: number;
  taxedAmount: number;
  interest: number;
  billStatus: string;
  dueDate: string;
  timeRemaining: number;
  interestExempt: boolean;
  archive: boolean;
}
